bundles/specmate-integration-test/src/com/specmate/test/integration/HistoryTest.java

```diff
@@ -15,69 +15,63 @@
 
 public class HistoryTest extends EmfRestTest {
 
-	public HistoryTest() throws Exception {}
-	
-	private String historyUrl(String... segments) {
-		return buildUrl("history", segments);
+	public HistoryTest() throws Exception {
 	}
-	
-	private String historyRecursiveUrl(String... segments) {
-		return buildUrl("historyRecursive", segments);
-	}
-	
-	private JSONArray getEntries(boolean recursive, String... segments) {
-		String historyUrl = recursive ? historyRecursiveUrl(segments) : historyUrl(segments);
-		RestResult<JSONObject> result = restClient.get(historyUrl);
+
+	private JSONArray getEntries(String type, String... segments) {
+		String historyUrl = buildUrl("history", segments);
+		RestResult<JSONObject> result = restClient.get(historyUrl, "type", type);
 		JSONObject history = result.getPayload();
 		return history.getJSONArray(HistoryPackage.Literals.HISTORY__ENTRIES.getName());
 	}
-	
+
 	/**
-	 * Tests that the history entries are returned is reversed creation order 
-	 * and expected state (creation, modification).
+	 * Tests that the history entries are returned is reversed creation order and
+	 * expected state (creation, modification).
 	 */
 	@Test
 	public void testHistoryIsInSequence() {
 		JSONObject requirement = postRequirementToRoot();
 		String requirementId = getId(requirement);
 		String newName = "changedName";
 		int numChangeNames = 11;
-		for(int i = 0; i < numChangeNames; i++) {
+		for (int i = 0; i < numChangeNames; i++) {
 			requirement.put(BasePackage.Literals.INAMED__NAME.getName(), newName + i);
 			updateObject(requirement, requirementId);
 		}
-		
-		checkSequence(getEntries(false, requirementId), newName, numChangeNames);
-		checkSequence(getEntries(true, requirementId), newName, numChangeNames);
+
+		checkSequence(getEntries("single", requirementId), newName, numChangeNames);
```

Comment:
Shouldn't this be the constants of HistoryRestService?

Reply:
Correct, fixed!